Orlando Walking Tours/Views/LocationTableViewCell.swift

```diff
@@ -7,11 +7,62 @@
 //
 
 import UIKit
+import Alamofire
 
-class LocationTableViewCell: UITableViewCell {
+class LocationTableViewCell: UITableViewCell, ReusableView
+{
+    ////////////////////////////////////////////////////////////
+    // MARK: - IBOutlets
+    ////////////////////////////////////////////////////////////
 
-    var locationId: String!
     @IBOutlet weak var locationThumbnail: UIImageView!
     @IBOutlet weak var locationTitle: UILabel!
     @IBOutlet weak var addLocationButton: UIButton!
+    @IBOutlet weak var loadingIndicator: UIActivityIndicatorView!
+
+    ////////////////////////////////////////////////////////////
+    // MARK: - Properties
+    ////////////////////////////////////////////////////////////
+
+    var locationId: String!
+    var request: Request?
+
+    ////////////////////////////////////////////////////////////
+    // MARK: - Helper Functions
+    ////////////////////////////////////////////////////////////
+
+    func configureImage(frame: CGRect)
+    {
+        reset()
+        loadImage(frame.width, height: frame.height)
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    func reset()
+    {
+        locationThumbnail.image = nil
+        request?.cancel()
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    func loadImage(width: CGFloat, height: CGFloat)
+    {
+        loadingIndicator.startAnimating()
+
+        // TODO: Image view for cell should be random photo of a location from the tour
+        request = UIImage.getPlaceholderImage(sized: Int(width), by: Int(height))
+        { image in
+            self.populateCell(image!)
+        }
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    func populateCell(image: UIImage)
```

Comment:
all UIImageViews accept UIImage? , add a ? to the end of the passed variable class.